cmd/config_helper.go

```diff
@@ -50,11 +50,21 @@ func toConfig(install, forceIgnored []*api.Package, targets []string, cmdline []
 	}
 
 	providers := collectProviders(forceIgnored, install)
-	packageNames := sortedKeys(allPackages)
-	sortedPackages := make([]*bazeldnf.RPM, 0, len(packageNames))
-	for _, name := range packageNames {
+	slices.Sort(targets)
+	sortedPackages := make([]*bazeldnf.RPM, 0, len(allPackages))
+
+	requested := make(map[string]bool)
+	for _, pkg := range targets {
+		requested[pkg] = true
+	}
+
+	// make sure all requested packages have their full dependency tree set
+	for _, name := range sortedKeys(allPackages) {
 		pkg := allPackages[name]
-		deps, err := collectDependencies(name, pkg.Dependencies, providers, ignored)
+		if _, requested := requested[name]; !requested {
```

Comment:
should use `ok` here instead of `requested` because that's the normal pattern and this overloads the variable name